vendor/processors/stripe_processor.py

```diff
@@ -198,6 +268,197 @@ def build_subscription(self, subscription, payment_method_id):
+            'trial_period_days': None if offer.term_details['trial_days'] < 1 else offer.term_details['trial_days'],
+            'metadata': {'site': offer.site.pk}
+
+        }
+
+        subscription = self.stripe_create_object(self.stripe.Subscription, subscription_data)
+        
+        return subscription
+
+    ##########
+    # Stripe & Vendor Objects Synchronization
+    ##########
+
+    def get_stripe_customers(self):
+        """
+        Returns all Stripe created customers
+        """
+        query = self.build_search_query([{
+            'key_name': 'site',
+            'key_value': self.site.pk,
+            'field_type': 'metadata'
+        }])
+        customer_search = self.stripe_query_object(self.stripe.Customer, {'query': query})
+        if customer_search:
+            return customer_search['data']
+        return None
+
+    def get_stripe_customers_in_vendor(self, customers):
+        """
+        Returns all vendor customers who have been created as Stripe customers
+        """
+        email_list = [customer_obj['email'] for customer_obj in customers]
+        stripe_users = CustomerProfile.objects.filter(user__email__in=email_list)
+        return stripe_users
+
+    def get_stripe_customers_not_in_vendor(self, customers):
+        """
+        Returns all vendor customers who have not been created as Stripe customers
+        """
+
+        email_list = [customer_obj['email'] for customer_obj in customers]
+        stripe_users = CustomerProfile.objects.exclude(user__email__in=email_list)
+        return stripe_users
+
+    def get_customers_to_check(self):
+        stripe_customers = self.get_stripe_customers()
+        stripe_customers_in_vendor = self.get_stripe_customers_in_vendor(stripe_customers)
+        stripe_customers_not_in_vendor = self.get_stripe_customers_not_in_vendor(stripe_customers)
+        return stripe_customers_in_vendor, stripe_customers_not_in_vendor
+
+    def sync_customers(self):
+        stripe_customers_in_vendor, stripe_customers_not_in_vendor = self.get_customers_to_check()
+
+        # Create customer objs for Vendor customers not in Stripe
+        for profile in stripe_customers_not_in_vendor:
+            profile_data = self.build_customer(profile)
+            new_stripe_customer = self.stripe_create_object(self.stripe.Customer, profile_data)
+            if new_stripe_customer:
+                profile.meta['stripe_id'] = new_stripe_customer['id']
+                profile.save()
+
+        # Update existing stripe customers info with vendor data
+        for profile in stripe_customers_in_vendor:
+            customer_id = profile.meta['id']
+            profile_data = self.build_customer(profile)
+            existing_stripe_customer = self.stripe_update_object(self.stripe.Customer, customer_id, profile_data)
+            if existing_stripe_customer:
+                profile.meta['stripe_id'] = existing_stripe_customer['id']
+                profile.save()
```

Comment:
This two loops should be their own function. That way you can have a much readble set of functions:
```
def sync_customer(self):
  stripe_customer = self.get_stripe_customer()
  stripe_customer_in_vendor = self.get_stripe_customers_in_vendor(stripe_customer)
  stripe_customer_not_in_vendor = self.get_stripe_customers_not_in_vendor(stripe_cusomters)

  self.create_stripe_customer(stripe_customer_not_in_vendor)

self.update_stripe_customer(stripe_customer_in_vendor)
```